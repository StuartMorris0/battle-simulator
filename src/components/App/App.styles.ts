import { motion } from 'framer-motion';
import styled from 'styled-components';

export const StyledWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`;

export const GameWrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 769px) {
    flex-direction: row;
  }
`;

export const ButtonWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 8px 20px;

  & button {
    width: 100%;
    background: #66bb6a;
    color: white;
    padding: 20px 40px;
    border: 1px solid white;
    outline: none;
    font-size: 22px;
    text-transform: uppercase;
    font-weight: bold;
    letter-spacing: -1px;

    @media (min-width: 760px) {
      width: auto;
    }

    &:hover {
      background-color: #3b903f;
    }
    &:disabled {
      background-color: #a9a9a9;
      border-color: #a9a9a9;
    }
  }
`;

export const TitleWrapper = styled(motion.div)`
  color: white;
  text-align: center;
  font-size: 32px;
  font-weight: bold;
  padding: 12px;
`;
