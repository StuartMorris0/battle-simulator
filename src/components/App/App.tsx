import React from 'react';

import GamePlayer from 'components/GamePlayer';
import { makeRollAsync } from 'features/game/gameSlice';
import { useAppDispatch, useAppSelector } from 'hooks/storeHooks';

import { GameWrapper, StyledWrapper } from './App.styles';

const App: React.FC = () => {
  const game = useAppSelector((state) => state.game);
  const dispatch = useAppDispatch();

  const handleOnClick = () => {
    dispatch(makeRollAsync());
  };

  return (
    <StyledWrapper>
      <GameWrapper>
        <GamePlayer player={game.player1} display="ltr" />
        <GamePlayer player={game.opponent} display="rtl" />
      </GameWrapper>
      <div>
        <button type="button" onClick={handleOnClick} disabled={game.status !== 'waiting'}>
          Attack
        </button>
      </div>
    </StyledWrapper>
  );
};

export default App;
