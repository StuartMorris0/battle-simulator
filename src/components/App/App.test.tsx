import React from 'react';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { userEvent } from 'global/utils/test-utils';
import { store } from 'store';

import App from '.';

describe('App Component', () => {
  const renderComponent = () =>
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  test('render snaphost', () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('handle roll and attack', async () => {
    jest.useFakeTimers();
    const { getByText, findByText } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    userEvent.click(getByText('Attack'));
    expect(await findByText('rolling...')).toBeInTheDocument();
    jest.runAllTimers();
  });
});
