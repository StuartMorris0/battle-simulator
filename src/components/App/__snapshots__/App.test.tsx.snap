// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component render snaphost 1`] = `
.c7 {
  display: block;
}

.c7 img {
  width: 50px;
  height: 50px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c11 {
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  font-weight: bold;
}

.c9 {
  background-color: white;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 3px;
  width: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 2px;
}

.c10 {
  background-color: #51da51;
  border-radius: 4px;
  display: block;
  width: 100%;
  height: 100%;
  -webkit-transition: height 1s ease-in-out;
  transition: height 1s ease-in-out;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  padding: 10px;
  position: relative;
}

.c5 {
  width: 50px;
  margin: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-box-flex: 0.5;
  -webkit-flex-grow: 0.5;
  -ms-flex-positive: 0.5;
  flex-grow: 0.5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  overflow: hidden;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  padding: 12px;
  height: 120px;
  width: 80px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100px;
  -webkit-flex: 0.8;
  -ms-flex: 0.8;
  flex: 0.8;
}

.c13 {
  font-size: 26px;
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  background: #505050;
  padding: 12px;
  margin: 8px 0px;
  text-align: center;
  box-shadow: none;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 8px 20px;
}

.c14 button {
  width: 100%;
  background: #66bb6a;
  color: white;
  padding: 20px 40px;
  border: 1px solid white;
  outline: none;
  font-size: 22px;
  text-transform: uppercase;
  font-weight: bold;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

.c14 button:hover {
  background-color: #3b903f;
}

.c14 button:disabled {
  background-color: #a9a9a9;
  border-color: #a9a9a9;
}

.c1 {
  color: white;
  text-align: center;
  font-size: 32px;
  font-weight: bold;
  padding: 12px;
}

@media (min-width:769px) {
  .c8 {
    height: 260px;
  }
}

@media (min-width:769px) {
  .c5 {
    width: 80px;
  }
}

@media (min-width:769px) {
  .c12 {
    height: auto;
  }
}

@media (min-width:769px) {
  .c0 {
    height: 100vh;
  }
}

@media (min-width:769px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .c2 .player:not(:first-child) {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:769px) {
  .c14 {
    margin: 40px 20px;
  }
}

@media (min-width:760px) {
  .c14 button {
    width: auto;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    style="opacity: 0; transform: scale(0.2) translateZ(0);"
  >
    BATTLE SIMULATOR
  </div>
  <div
    class="c2"
  >
    <div
      class="c3 player"
      data-testid="GamePlayer"
    >
      <div
        class="c4"
        data-testid="PlayerAvatar"
      >
        <img
          alt="PLAYER"
          class="c5"
          src="player.svg"
        />
        <span>
          PLAYER
        </span>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-testid="RolledDice"
        >
          <img
            alt="dice"
            src="blank.svg"
          />
        </div>
        <div
          class="c7"
          data-testid="RolledDice"
        >
          <img
            alt="dice"
            src="blank.svg"
          />
        </div>
      </div>
      <div
        class="c8"
        data-testid="HealthBar"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
          />
        </div>
        <div
          class="c11"
        >
          100
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <span
        class="c13"
      >
        Tap ATTACK to start...
      </span>
    </div>
    <div
      class="c3 player"
      data-testid="GamePlayer"
    >
      <div
        class="c4"
        data-testid="PlayerAvatar"
      >
        <img
          alt="MONSTER"
          class="c5"
          src="monster.svg"
        />
        <span>
          MONSTER
        </span>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-testid="RolledDice"
        >
          <img
            alt="dice"
            src="blank.svg"
          />
        </div>
        <div
          class="c7"
          data-testid="RolledDice"
        >
          <img
            alt="dice"
            src="blank.svg"
          />
        </div>
      </div>
      <div
        class="c8"
        data-testid="HealthBar"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
          />
        </div>
        <div
          class="c11"
        >
          100
        </div>
      </div>
    </div>
  </div>
  <div
    class="c14"
  >
    <button
      type="button"
    >
      Attack
    </button>
  </div>
</div>
`;
